vignettes/datatable-programming.Rmd

```diff
@@ -25,8 +25,14 @@ knitr::opts_chunk$set(
 
 `data.table`, from its very first releases, enabled the usage of `subset` and `with` (or `within`) functions by defining the`[.data.table` method. `subset` and `with` are base R functions that are useful for reducing repetition in code, enhancing readability, and reducing number the total characters the user has to type. This functionality is possible in R because of a quite unique feature called *lazy evaluation*. This feature allows a function to catch its arguments, before they are evaluated, and to evaluate them in a different scope than the one in which they were called. Let's recap usage of the `subset` function.
 
-```{r opt_max_print_10, include = FALSE}
-options(max.print = 10L) # 2 rows
+```{r df_print, echo=FALSE}
+registerS3method("print", "data.frame", function(x, ...) {
+  base::print.data.frame(head(x, 2L), ...)
+  cat("...\n")
+  invisible(x)
+})
+.topn = options("datatable.print.topn"=2L)
```

Comment:
nit: I usually prefer unquoted keyword arg when possible, i.e. `options(datatable.print.topn=2L)`. Feel free to leave be.

less nit: call `options()` just once:

```r
old <- options(
  datatable.print.topn=2L,
  datatable.print.nrows=20L
)

# below
options(old)
```